File: Microservices/IoT.House.Automation.Microservices.Arduino/IoT.House.Automation.Microservices.Arduino.Application/MessageBroker/Handlers/ArduinoAddedEventHandler.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using IoT.House.Automation.Microservices.Arduino.Application.Interfaces;
using IoT.House.Automation.Microservices.Arduino.Application.MessageBroker.Events;

namespace IoT.House.Automation.Microservices.Arduino.Application.MessageBroker.Handlers
{
    public class ArduinoAddedEventHandler : IEventHandler<ArduinoAddedEvent>
    {
        public Task Handle(ArduinoAddedEvent @event)
        {
            var ran = new Random().Next(1, 100);

            return ran < 50 ? Task.CompletedTask : Task.FromException(new NotImplementedException());
        }
    }
}

File: Microservices/IoT.House.Automation.Microservices.Mobile/sync/src/main/java/iot/house/automation/microservices/mobile/sync/handlers/ArduinoAddedEventHandler.java
package iot.house.automation.microservices.mobile.sync.handlers;

import iot.house.automation.microservices.mobile.application.interfaces.EventHandler;
import iot.house.automation.microservices.mobile.application.messaging.events.ArduinoAddedEvent;
import iot.house.automation.microservices.mobile.domain.interfaces.ArduinoManagement;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.Future;

@Component
public class ArduinoAddedEventHandler implements EventHandler<ArduinoAddedEvent> {

    private final ArduinoManagement arduinoManagement;

    @Autowired
    public ArduinoAddedEventHandler(ArduinoManagement arduinoManagement) {
        this.arduinoManagement = arduinoManagement;
    }

    @Override
    public Future handle(ArduinoAddedEvent event) {
        if (event == null) return CompletableFuture.completedFuture(true);

        var arduino = event.getArduino();

        try {
            this.arduinoManagement.register(arduino);
            return CompletableFuture.completedFuture(true);
        } catch (Exception e) {
            return CompletableFuture.failedFuture(e);
        }
    }
}
